### ISSUE TITLE
Error in  "TypeError: descriptor 'intersection' for 'set' objects doesn't apply to a 'str' object"

### ISSUE THREAD

**Issue by 20261119** [KennthShang/PhaBOX#32]
### This is my  script :
phabox2 --task tree  --dbdir /hl/Liuzifeng/database/phabox_db --outpth vOTU_20kb --msa Y --tree Y  --contigs vOTU_20kb.fa --threads 40
PhaBOX2 is running with: 40 threads!
Running program: marker gene searching
[1/5]filtering the length of contigs...
[2/5] reusing existing protein file...
[3/5] running marker:
Current marker: terl...
Current marker: portal...
[4/5] Running the msa...
Running msa: terl...This may take a long time...
Running msa: portal...This may take a long time...
Traceback (most recent call last):
  File "/hl/Liuzifeng/miniconda3/envs/phabox2/bin/phabox2", line 10, in <module>
    sys.exit(main())
  File "/hl/Liuzifeng/miniconda3/envs/phabox2/lib/python3.10/site-packages/phabox2/phabox2.py", line 407, in main
    tree.run(inputs)
  File "/hl/Liuzifeng/miniconda3/envs/phabox2/lib/python3.10/site-packages/phabox2/tree.py", line 139, in run
    common = set.intersection(*common)
TypeError: descriptor 'intersection' for 'set' objects doesn't apply to a 'str' object

**Comment by KennthShang** [comment]
Hi @20261119 ,

Thanks for catching this. However, I am on leave today. Will update to you as soon as possible.

BTW, you can use the .aln file in the vOTU_20kb/midfolder/msa and combine them yourself if you are in a hurry.

Sorry for the inconvenient.

Best,
Jiayu

**Comment by 20261119** [comment]
ok,thank you. I will try it

**Comment by KennthShang** [comment]
Hi @20261119 ,

The issue is fixed now. Please run the following codes to updates your phabox2:


```

git clone https://github.com/KennthShang/PhaBOX.git
cd PhaBOX


conda activate phabox2
python -m pip install . -vv

# test
phabox2 --help

# download the latest database
wget https://github.com/KennthShang/PhaBOX/releases/download/v2/phabox_db_v2.zip
unzip phabox_db_v2zip > /dev/null
```

Then, you can rerun your codes. If you do not prefer the FastTree or the FastTree still have a problem, you can use the combiend.msa in the final_prediction folder for further tree building process.

BTW, the metadata of the reference genomes/genes can be found in phabox_db_v2/marker/marker_stats.tsv


Hope this information will help.

Best,
Jiayu

**Comment by 20261119** [comment]
Thanks, I'll run it later.

**Comment by KennthShang** [comment]
Sorry that I found another issue in the tree construction. The previous codes will result in DNA sequences rather than protein sequences. New release will be update tomorrow morning once PhaBOX2 pass my local test.

Sorry again for the inconvenience. If you have any usage problems, you can also add my WeChat: KennethShang, but do not forget to add some description (from GitHub and your GitHub ID) for the request.

Best,
Jiayu

**Comment by KennthShang** [comment]
The issue is fixed in v2.1.5.

Best,
Jiayu
